gap/attr.gi

```diff
@@ -1524,6 +1524,26 @@ function(D)
   return Concatenation(loops, out);
 end);
 
+# Compute all undirected simple circuits by filtering the output
+# of DigraphAllSimpleCircuits
+InstallMethod(DigraphAllUndirectedSimpleCircuits, "for a digraph",
+                     [IsDigraph], function(D)
+    local digraph, cycles, remove, cycle, cycleRev;
+    digraph := DigraphSymmetricClosure(
+        DigraphRemoveAllMultipleEdges(DigraphMutableCopyIfMutable(D)));
+    cycles := Filtered(DigraphAllSimpleCircuits(digraph),
+        c -> Length(c) > 2 or Length(c) = 1);
+    remove := [];
+    for cycle in cycles do
+        if not cycle in remove and Length(cycle) <> 1 then
+            cycleRev := [cycle[1]];
+            Append(cycleRev, Reversed(cycle{[2 .. Length(cycle)]}));
+            Add(remove, cycleRev);
+        fi;
+    od;
+    return Difference(cycles, remove);
```

Comment:
Perhaps you could try flipping the logic so that instead of storing cycles to remove, the function stores cycles to keep, this is a small optimization and would avoid having a call to `Difference` at the end. So something like
```
keep := []
for cycle in cycles do
  cycleRev := [cycle[1]];
  Append(cycleRev, Reversed(cycle{[2 .. Length(cycle)]}));
  if (not cycle in keep and not cycleRev in keep) or Length(cycle) <> 1 then
    Add(keep, cycle);
  fi;
od;
return keep;
```
might work.